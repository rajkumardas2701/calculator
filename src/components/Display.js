import React from 'react';
import PropTypes from 'prop-types';

export default function Display({ total, operation, next }) {
  return (
    <div className="display">
      {total}
      {' '}
      {operation}
      {' '}
      {next}
    </div>
  );
}

Display.defaultProps = {
  total: '0',
  operation: null,
  next: null,
};

Display.propTypes = {
  total: PropTypes.string,
  operation: PropTypes.string,
  next: PropTypes.string,
};
