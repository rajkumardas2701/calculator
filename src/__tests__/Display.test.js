import React from 'react';
import { render, cleanup, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Display from '../components/Display';

afterEach(cleanup);

it('displays the current result', () => {
  const { renderer } = render(<Display total="10" />);
  expect(screen.getByTestId('test-display')).toHaveTextContent('10');
  expect(renderer).toMatchSnapshot();
});
