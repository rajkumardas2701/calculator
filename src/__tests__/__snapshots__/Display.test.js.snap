// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the current result 1`] = `undefined`;
